### PR TITLE
Raul add owner message section to the top

### PR DESCRIPTION
**Pull request by rauleffting**
# Description
(DASHBOARD COMPONENT - PRIORITY LOW) 

> (PRIORITY LOW) Jae: Add to top of app a section that the “Owner” class (only) can edit to post a message to all users of the app like, “Happy Holidays”. (WIP - Raul)
> When not showing a custom message, it should show our horizontal logo. Give me the dimensions I’m allowed and I’ll have our graphics guy create one just for this.
> I’d like the UI for posting the message to have the ability for me to set an expiration date so it’ll automatically return to the logo on that date

## Mainly changes explained:
An owner message section was added to the Header. So now, owner accounts can post messages by image or text on the header for all the volunteers.

## How to test:
1. this PR is related to backend [PR #288](https://github.com/OneCommunityGlobal/HGNRest/pull/288). You have to use both to test this feature.
2. git checkout raul_add_owner_section_to_the_top
3. npm install
4. npm run start:local
5. log in as **OWNER**
6. A pen should be displayed for you:
header

7. click and try to create and delete messages. For image messages, try using this one - which the resolution is made for this resource:
image

8. The pen should only be displayed for owner accounts. The message has to be displayed for everyone.
9. This resource is made only for computers (width higher than 1400px).

### CODE REVIEW

**Comment by nicollext** [comment]
Tested with the as a Owner and it is working as expected. 

[https://www.dropbox.com/home/Highest%20Good%20Network%20Team/Nicolle/Week%202/PR%20%23288%20-%20raul_add_owner_section_to_the_top?preview=raul_add_owner_section_to_the_top.mp4](url)

[https://www.dropbox.com/home/Highest%20Good%20Network%20Team/Nicolle/Week%202/PR%20%23288%20-%20raul_add_owner_section_to_the_top?preview=raul_add_owner_section_to_the_top_2.mp4](url)

The digital pen also  is not visible on the screen for Volunteers who are logged in.

[https://www.dropbox.com/s/m4ugfb1boxe25hj/raul_add_owner_section_to_the_top_3.PNG?dl=0](url)

**Comment by crystalsong2021** [comment]
Hi @rauleffting, I tested the UI for adding custom message for the user. 
I am able to create the message by clicking the icon, add the message, add the picture.
I did noticed when my window is smaller, I could not locate the icon. I shown it in the video. Another thing I noticed is that the message did not shown correctly on volunteer account. It did shown correctly on the admin account. 
https://user-images.githubusercontent.com/18197013/223592919-3cb12a54-8bae-4a2f-a61c-ea9a540a783e.mp4

**Review by Lucas-E — DISMISSED** [review]
Raul, I have reviewed your PR and I must say, it works really well. I observed that the message is displayed on both the volunteer and administrator accounts without the icon. great job!
Captura de tela de 2023-03-07 22-57-26
Captura de tela de 2023-03-07 22-57-04
Captura de tela de 2023-03-07 22-53-02

File: src/components/OwnerMessage/OwnerMessage.jsx [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+            style={message ? { display: 'block' } : { display: 'none' }}
+          >
+            Delete
+          </Button>
+          <Button color="primary" onClick={handleMessage}>
+            {message ? 'Update' : 'Create'}
+          </Button>
+        </ModalFooter>
+      </Modal>
+      <Modal isOpen={modalDeleteWarning} toggle={toggleDeleteWarning}>
```
Comment (crystalsong2021) [inline src/components/OwnerMessage/OwnerMessage.jsx:162]: good job setting the error boundary message and the picture type.

**Review by crystalsong2021 — COMMENTED** [review]
Nice job working on the owner message section at top bar. If you could adjust the code to show the writing icon in smaller window size, that will be great. Thank you.

**Comment by rauleffting** [comment]
> Hi @rauleffting, I tested the UI for adding custom message for the user. I am able to create the message by clicking the icon, add the message, add the picture. I did noticed when my window is smaller, I could not locate the icon. I shown it in the video. Another thing I noticed is that the message did not shown correctly on volunteer account. It did shown correctly on the admin account. https://user-images.githubusercontent.com/18197013/223592919-3cb12a54-8bae-4a2f-a61c-ea9a540a783e.mp4

Hi, @crystalsong2021! No problem. It's supposed to behave like this. After the approval, I'm going to add this as a new bug. As it's a low-priority issue and very time-consuming to do, it will be left open to someone else. Thank you very much!

**Comment by vitorsaa2k** [comment]
Hello @rauleffting,

Such a great feature to have on the app.

I ran into a couple of inconcistences when testing, as showed in the video:

https://user-images.githubusercontent.com/17365161/223816461-39b017c2-302b-4e31-ae0d-710eb3318d4c.mp4

Sometimes I make changes and it wont update until I relog in another account, sometimes it updates when I reaload the page.
And looks like its more difficult to make it update for the others users when using the "update" button on the modal. When using the "create" button it seems to work more easily.

**Comment by rauleffting** [comment]
> Hello @rauleffting,
> 
> Such a great feature to have on the app.
> 
> I ran into a couple of inconcistences when testing, as showed in the video:
> 
> Sometimes I make changes and it wont update until I relog in another account, sometimes it updates when I reaload the page. And looks like its more difficult to make it update for the others users when using the "update" button on the modal. When using the "create" button it seems to work more easily.

Hi Vitor. About the update, this is expected because of the testing server. It's very slow.

**Comment by rauleffting** [comment]
Hi guys. I fixed some conflicts with Jest - we are not using it yet. If you could approve it again, I'll be very thankful. Thanks!

**Review by crystalsong2021 — DISMISSED** [review]
Thank you for taking the time to add the feature and planning the logic.

**Review by NyahJ — APPROVED** [review]
Hi @rauleffting,

Great job! Codes are clear and the functionality is going well as expected, so I am gonna approve this, thanks!
截屏2023-03-08 下午7 02 56

**Comment by PrabhjotSembhi** [comment]
Hey, @rauleffting I can see that message on top but I was not able to log in as Owner. I am not seeing that option
image

I have tried everything multiple times but still cant seeing option to switch role as Owner

**Comment by rauleffting** [comment]
> Hey, @rauleffting I can see that message on top but I was not able to log in as Owner. I am not seeing that option image
> 
> I have tried everything multiple times but still cant seeing option to switch role as Owner

Hi @PrabhjotSembhi. Please, use postman to create an owner account.

**Comment by one-community** [comment]
You can also PM me with the email of one of your Dev accounts you want changed to "owner" and I'll update it for you

**Comment by Jorge-Nerd** [comment]
Hello @rauleffting  good job it work fine for me
image

**Comment by one-community** [comment]
Thank you everyone, moving this to final review.

**Pull request merged**